digitaltwins/samples/model_repository/search_model.js

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+const ModelRepositoryCredentials = require('azure-iot-digitaltwins-model-repository').ModelRepositoryCredentials;
+const ModelRepositoryServiceClient = require('azure-iot-digitaltwins-model-repository').ModelRepositoryServiceClient;
+
+// Simple example of how to:
+// - create a Model Repository Service Client
+// - authenticate Service Client with Model Repository
+// - search for Digital Twin Models in the Model Repository
+// - use search keywords and filters
+async function main() {
+  // Azure IoT Model repository connection string has to be set to system environment variable AZURE_IOT_MODEL_REPOSITORY_CONNECTION_STRING
+
+  const modelRepositoryCredentials = new ModelRepositoryCredentials(process.env.AZURE_IOT_MODEL_REPOSITORY_CONNECTION_STRING);
```

Comment:
From my quick read as a C dev reading Node, it seems that the MODEL_REPOSITORY_CONNECTION string is always required but then later in the options we let the 'repositoryId' specify whether we hit global repository or private?  Or can the process.env.AZURE thingy be NULL and we fall to global? - though if yes feels this may need a comment.

Should we have a path where we can just search global without need for this env?

Reply:
My understanding:
- We always need a connection string to authenticate the Service Client. 
- After the Client has been authenticated we use the RepositoryId for private repository access only.
I learned it from the generated code comments.
So, I think this is by design.

---
In reply to: [338375454](https://github.com/Azure/azure-iot-sdk-node/pull/646#discussion_r338375454) [](ancestors = 338375454)